lib/ox/element.rb

```diff
@@ -150,7 +150,18 @@ def locate(path)
     # _raise_ [NoMethodError] if no match is found
     def method_missing(id, *args, &block)
       has_some = false
-      ids = id.to_s
+      ids = id.to_s.sub('_', ':') # allows easy API with namespaces
+      
+      if ids[-1] == '!' # return array of elements matching ids
```

Comment:
Probably best to support both Symbol and String when comparing to the node name.

The second form would be a Hash where the keys would match against Element#attributes. So ```.each(foo: 'bar')``` would match any child element that has an attribute ```foo``` with a value of ```bar```.